src/sbd-inquisitor.c

```diff
@@ -1307,38 +1303,52 @@ int main(int argc, char **argv, char **envp)
         } else if (strcmp(argv[optind], "ping") == 0) {
             exit_status = ping_via_slots(argv[optind + 1], servants_leader);
 
-        } else if (strcmp(argv[optind], "watch") == 0) {
+        } else if (strcmp(argv[optind], "query-watchdog") == 0) {
+            exit_status = watchdog_info();
+
+        } else if (strcmp(argv[optind], "test-watchdog") == 0) {
+            exit_status = watchdog_test();
+
+        } else {
+            exit_status = -2;
+        }
+#endif
+
+        if (strcmp(argv[optind], "watch") == 0) {
+            /* sleep $(sbd $SBD_DEVICE_ARGS dump | grep -m 1 msgwait | awk '{print $4}') 2>/dev/null */
+
+#if SUPPORT_SHARED_DISK
                 if(disk_count > 0) {
                     /* If no devices are specified, its not an error to be unable to find one */
                     open_any_device(servants_leader);
                 }
+#endif
+                /* Re-calculate timeout_watchdog_warn based on any timeout_watchdog from:
+                 * SBD_WATCHDOG_TIMEOUT, -1 option or on-disk setting read with open_any_device() */
+                if (do_calculate_timeout_watchdog_warn) {
+                    timeout_watchdog_warn = calculate_timeout_watchdog_warn(timeout_watchdog);
+                }
+
+                cl_log(LOG_DEBUG, "Delay start: %s%s%s",
```

Comment:
why don't we postpone that to after we've actually calculated the values that are being executed so that we can display them.
to simplify the logic and still not loose the information where the data came from we can set a pointer to a string constant when updating the delay from e.g. 'SBD_WATCHDOG_TIMEOUT * 2', 'msgwait', 'watchdog-timeout from cmdline * 2' (possible other source of watchdog would be the disk-header but this isn't relevant here as we would the go for msgwait taken from there) and add that as a hint to the log.